api/migrations/0003_fake_data.py

```diff
@@ -0,0 +1,284 @@
+from datetime import timedelta, time
+
+from django.db import migrations, models
+from django.db.backends.sqlite3.schema import DatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.db.models import Q
+from faker import Faker
+from model_bakery.recipe import Recipe, foreign_key, related
+
+from api.models import (
+    AgeRange,
+    DayAndTimeSlot,
+    NonTeachingHelp,
+    LanguageAndLevel,
+    Language,
+)
+from api.models.choices.age_range_type import AgeRangeType
+from api.models.choices.communication_language_mode import CommunicationLanguageMode
+from api.models.choices.registration_telegram_bot_language import (
+    RegistrationTelegramBotLanguage,
+)
+from api.models.choices.statuses import (
+    CoordinatorStatus,
+    StudentStatus,
+    TeacherStatus,
+    TeacherUnder18Status,
+    GroupStatus,
+)
+from api.models.data_migration_maker import DataMigrationMaker
+
+APP_NAME = "api"
+
+AMOUNT_OF_COORDINATORS_WITHOUT_GROUP = 10
+AMOUNT_OF_GROUPS = 30
+AMOUNT_OF_STUDENTS_WITHOUT_GROUP = 30
+AMOUNT_OF_TEACHERS_WITHOUT_GROUP = 5
+AMOUNT_OF_TEACHERS_UNDER_18_WITHOUT_SPEAKING_CLUB = 5
+AMOUNT_OF_SPEAKING_CLUBS = 7
+
+
+class FakeDataPopulator(DataMigrationMaker):
+    def __init__(self, apps: StateApps, schema_editor: DatabaseSchemaEditor) -> None:
+        super().__init__(apps, schema_editor)
+        self.faker: Faker = Faker()
+        self.personal_info_recipe = self._make_personal_info_recipe()
+        self.coordinator_recipe = self._make_coordinator_recipe()
+        self.student_recipe = self._make_student_recipe()
+        self.teacher_recipe = self._make_teacher_recipe()
+        self.teacher_under_18_recipe = self._make_teacher_under_18_recipe()
+        self.group_recipe = self._make_group_recipe()
+        self.speaking_club_recipe = self._make_speaking_club_recipe()
+
+    def _get_random_amount_of_objects(
+        self,
+        model: type[models.Model],
+        filter_condition: Q = Q(),
+        min_length: int = 0,
+        max_length: int | None = None,
+    ) -> models.QuerySet:
+        """Returns a random amount of objects from the database as a queryset."""
+        queryset = model.objects.filter(filter_condition)
+        if max_length is None:
+            max_length = queryset.count()
+        return self.faker.random_choices(
+            queryset,
+            length=self.faker.pyint(min_value=min_length, max_value=max_length),
+        )
+
+    def _make_group_common_recipe(
+        self, model_name: str, teachers_min_value: int, teachers_max_value: int
+    ) -> Recipe:
+        return Recipe(
+            model_name,
+            coordinators=lambda: related(
+                *[self.coordinator_recipe] * self.faker.pyint(min_value=1, max_value=5)
+            ),
```

Comment:
could you explain in a short comment in code what this `related` with the unpacking does here?

Reply:
I've tried :)

Reply:
Reply to the above comment: I've split logic into two classes, made an RecipeStorage class which stores recipes